aziot/src/internal/check/checks/host_local_time.rs

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+use super::prelude::*;
+
+#[derive(Serialize, Default)]
+pub struct HostLocalTime {
+    offset: Option<i64>,
+}
+
+#[async_trait::async_trait]
+impl Checker for HostLocalTime {
+    fn meta(&self) -> CheckerMeta {
+        CheckerMeta {
+            id: "host-local-time",
+            description: "host time is close to reference time",
+        }
+    }
+
+    async fn execute(&mut self, shared: &CheckerShared, cache: &mut CheckerCache) -> CheckResult {
+        self.execute_inner(shared, cache)
+            .await
+            .unwrap_or_else(CheckResult::Failed)
+    }
+}
+
+impl HostLocalTime {
+    async fn execute_inner(
+        &mut self,
+        shared: &CheckerShared,
+        _cache: &mut CheckerCache,
+    ) -> Result<CheckResult> {
+        // TODO: check against parent time in nested edge scenarios
+
+        fn is_server_unreachable_error(err: &mini_sntp::Error) -> bool {
+            match err {
+                mini_sntp::Error::ResolveNtpPoolHostname(_) => true,
+                mini_sntp::Error::SendClientRequest(err)
+                | mini_sntp::Error::ReceiveServerResponse(err) => {
+                    err.kind() == std::io::ErrorKind::TimedOut || // Windows
```

Comment:
Can safely remove Windows support here. Unless this is needed by WSL?

Reply:
https://github.com/hyperium/hyper/issues/626#issuecomment-130754048

Since we aren't targeting windows anymore, sure, it'd be fine to remove this. At the same time, this single line allows `mini_sntp` to run on windows, should the need ever arise. Personally, I'd just leave it in...

Reply:
Sure we can leave it in